File: ios/Classes/Tasks/Task.m
//
//  Task.m
//  geolocator
//
//  Created by Maurits van Beusekom on 20/05/2019.
//

#import "Task.h"

@implementation Task

- (instancetype)initWithContext:(TaskContext *)context completionHandler:(CompletionHandler)completionHandler {
    self = [super init];
    if (self) {
        self.context = context;
        self.completionHandler = completionHandler;
        self.taskID = [[NSUUID alloc] init];
    }
    
    return self;
}

- (void)stopTask {
    if (_completionHandler != nil) {
        _completionHandler(_taskID);
    }
}

- (void)handleErrorCode:(NSString *)code message:(NSString *)message {
    _context.resultHandler([FlutterError errorWithCode:code message:message details:nil]);
}

@end

File: ios/Classes/Tasks/Task.swift
//
//  Task.swift
//  geolocator
//
//  Created by Maurits van Beusekom on 18/07/2018.
//

import Foundation

class Task: NSObject {
    required init(context: TaskContext,
                  completionHandler: CompletionHandler?) {
        
        self.taskID = UUID.init()
        self.context = context
        self.completionHandler = completionHandler
    }
    
    let taskID: UUID
    let context: TaskContext
    let completionHandler: CompletionHandler?
    
    func stopTask() {
        guard let action = completionHandler else { return }
        action(taskID)
    }
    
    func handleError(code: String, message: String) {
        self.context.resultHandler(FlutterError.init(
            code: code,
            message: message,
            details: nil))
    }
    
}
